tests/sqlparser_postgres.rs

```diff
@@ -4441,3 +4441,614 @@ fn test_table_unnest_with_ordinality() {
         _ => panic!("Expecting TableFactor::UNNEST with ordinality"),
     }
 }
+
+#[test]
+fn parse_create_trigger() {
+    for referencing in [
```

Comment:
I'm a bit worried the multiple nestings make this test somewhat harder to follow, and as a result make changes to, going forward. To simplify we could have one or few asserts that looks at the body of the statement.
Then for tests on the diffferent variations of the various components of the statement, we can rely on shorter roundtrip-only assertions - here's one example test that does similar https://github.com/sqlparser-rs/sqlparser-rs/blob/e81a58cd88a55db26f16c035aac04ed4e10c1c59/tests/sqlparser_common.rs#L7288-L7337

Reply:
Hi - I am not sure I understood. The number of things to test is combinatorial, so, naturally, there will be lots of nested loops for all of the combinations, right? I believe it is much better to have nested for loops than extremely repetitive tests because as it is right now one can edit a single thing and be done with it.

I have not understood how the example you provided handles combinatoriality.

Reply:
Ah so it was test scenarios in this style specifically I was trying to highlight in the link
https://github.com/sqlparser-rs/sqlparser-rs/blob/e81a58cd88a55db26f16c035aac04ed4e10c1c59/tests/sqlparser_common.rs#L7325-L7330

maybe to use a clearer example, the test right before the test in question https://github.com/sqlparser-rs/sqlparser-rs/blob/c3a96e0be51b7057e6f597fc56eab1876fe6a1c4/tests/sqlparser_postgres.rs#L4350-L4387

so e.g. just calling

```sql
pg_and_generic().verified_expr("CURRENT_TIMESTAMP AT TIME ZONE tz");
```
as a test case can be similarly effective in demonstrating the parser behavior, given that we already have other test case(s) with more detailed asserts on the actual returned AST body for the statement/expr type being tested.

Re the combinatorial aspects of the syntax, a for loop could be reasonable when there are 1-2 variables, but this case feels extreme (has more cons than pros) to me so that it makes sense to me to consider the alternative style - my thinking is also that, when reading the code, if one needs to see that a specific syntax is supported, its clear from the list of raw sql what's supported vs not. Similarly, if one needs to include a new test case for a combination/new-variant, they only need to add one-line of code. In both cases, it would be a hassle to have to figure out the layers of nesting and mechanism this specific test uses in order to find what they're looking for, or to find if/where to inject their new test case.

Above is subjective ofc, I speak for only myself that the current func is harder for me to tell what's tested vs what's not - and the function sticks out when compared to others in the codebase (including those with more complicated syntax).

Reply:
Ok, what about keeping both the combinatorial loop and adding more examples of parsable strings such as the ones you propose? This way there is both the complete testing of all possible branches and combinations, plus the examples that a user may CTRL+F.

Reply:
The combinatorial loop is primarily the source of complexity in my mind and would ideally be flattened in order to simplify it.

Reply:
I have flattened and tried to clean as much the loop, and I have now included more cases like the CONSTRAINT keyword and the `FROM {referenced_table_name}` case. This covers all possible combinations of the CREATE TRIGGER, and the combinatoriality led me to discover a few cases I had not covered in the parser adequately. I hope this is now more readable, do let me know.

Reply:
Hmm, I think the situation didn't improved with the latest changes unfortunately, if anything the custom iterating obfuscates further compared to a familiar for-loop. By flattening I meant as in my previous comments/examples on this thread - is it something preventing us from using the same style as the linked examples for this syntax (I imagine there are plenty examples with that style in the codebase to follow)?

Reply:
Those approaches simply do not cover all possible combinations that are relevant to these objects. Yesterday, extending the combinations further, I found corner cases that I would have not thought of otherwise.

Writing out extensively the tests with the approach you have suggested, **as I understand it,** would require writing an extremely large amount of tests, which would ultimately be less readable than the current simple for loop.